ScoutSuite/providers/aws/services/ec2/instances.py

```diff
@@ -0,0 +1,31 @@
+from ScoutSuite.providers.aws.resources.resources import AWSResources
+from ScoutSuite.providers.aws.facade.facade import AWSFacade
+from ScoutSuite.providers.aws.aws import get_name
+from ScoutSuite.providers.aws.utils import ec2_classic, get_keys
+
+
+class EC2Instances(AWSResources):
+    async def fetch_all(self, **kwargs):
+        raw_instances  = self.facade.ec2.get_instances(self.scope['region'], self.scope['vpc'])
+        for raw_instance in raw_instances:
+            name, resource = self._parse_instance(raw_instance)
+            self[name] = resource
+
+    def _parse_instance(self, raw_instance):
+        instance = {}
```

Comment:
I think we got an opportunity with this migration to be PEP 8 compliant and start defining dictionaries like the following:

```python
        instance = {
                'id' = raw_instance['InstanceId'],
                'reservation_id' = raw_instance['ReservationId'],
                'monitoring_enabled' = raw_instance['Monitoring']['State'] == 'enabled',
                'user_data' = self.facade.ec2.get_instance_user_data(self.scope['region'], id)
        }
```

Reply:
I talked about it with Rémi last Thursday, and we came to the conclusion it's probably a better idea to continue with the current implementation. There were a couple of reasons why, but here are the most important IMHO:

1. *Easier debugging*: let's say raw_instance['InstanceId'] fails because the dict does not have this key. Using the PEP 8 convention, we would not know exactly what caused the crash.
2. Sometimes, you want to add stuff iteratively. That's not possible with the PEP 8 convention unless you want to use a hybrid of the two notations, but then we might as well keep up with the current solution